lib/vtls/mbedtls.c

```diff
@@ -444,6 +444,16 @@ mbed_connect_step1(struct connectdata *conn,
   mbedtls_debug_set_threshold(4);
 #endif
 
+  /* give application a chance to interfere with mbedTLS set up. */
+  if(data->set.ssl.fsslctx) {
+    ret = (*data->set.ssl.fsslctx)(data, &connssl->config,
+                                   data->set.ssl.fsslctxp);
+    if(ret) {
+      failf(data, "error signaled by mbedTLS ctx callback");
+      return CURLE_SSL_CERTPROBLEM;
```

Comment:
[CURLOPT_SSL_CTX_FUNCTION](https://curl.haxx.se/libcurl/c/CURLOPT_SSL_CTX_FUNCTION.html) says "If an error is returned from the callback no attempt to establish a connection is made _and the perform operation will return the callback's error code_."

Also saying it's mbedTLS ctx callback is confusing, it's the user's callback. `mbedTLS: ctx callback` would be less but rather just say ssl like we do in openssl.c, use [these two lines](https://github.com/curl/curl/blob/curl-7_52_1/lib/vtls/openssl.c#L2129-L2130)